packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.test.ts

```diff
@@ -0,0 +1,1396 @@
+        start: getFutureDateUTC(14, TimeOfDay.START),
+        end: getFutureDateUTC(16, TimeOfDay.END),
+        notes: "",
+        reasonId: 1,
+        toUserId: users[3].id,
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
+      },
+      {
+        userId: users[2].id,
+        uuid: uuidv4(),
+        start: getFutureDateUTC(18, TimeOfDay.START),
+        end: getFutureDateUTC(18, TimeOfDay.END),
+        notes: "",
+        reasonId: 2,
+        toUserId: users[0].id,
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
+      },
+      {
+        userId: users[3].id,
+        uuid: uuidv4(),
+        start: getFutureDateUTC(20, TimeOfDay.START),
+        end: getFutureDateUTC(20, TimeOfDay.END),
+        notes: "",
+        reasonId: 1,
+        toUserId: users[1].id,
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
+      },
+      {
+        userId: users[3].id,
+        uuid: uuidv4(),
+        start: getFutureDateUTC(22, TimeOfDay.START),
+        end: getFutureDateUTC(22, TimeOfDay.END),
+        notes: "",
+        reasonId: 2,
+        toUserId: users[0].id,
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
+      },
+    ],
+  });
+}
+
+describe("outOfOfficeHandler", () => {
+  setupAndTeardown();
+
+  describe("outOfOfficeCreateOrUpdate", () => {
+    it("should successfully create outofoffice for logged-in user.", async () => {
+      const users = [
+        {
+          id: 1,
+          name: "User 1",
+          username: "user-1",
+          email: "user-1@test.com",
+          timeZone: "Asia/Kolkata",
+          schedules: [TestData.schedules.IstWorkHours],
+        },
+      ];
+      addUsers(users);
+
+      const { schedules, ...loggedInUser } = users[0];
+      const ctx = {
+        user: loggedInUser as NonNullable<TrpcSessionUser>,
+      };
+
+      const input: TOutOfOfficeInputSchema = {
+        dateRange: {
+          startDate: new Date("2024-10-01T00:00:00.000Z"),
+          endDate: new Date("2024-10-01T00:00:00.000Z"),
+        },
+        offset: 330,
+        toTeamUserId: null,
+        reasonId: 1,
+      };
+
+      const result = await outOfOfficeCreateOrUpdate({ ctx, input });
+      expect(result).toStrictEqual({});
+
+      //expect outOfOffice to be created in db for userId:1
+      const ooo = await prismock.outOfOfficeEntry.findMany({
+        where: {
+          userId: users[0].id,
+        },
+      });
+      expect(ooo.length).toEqual(1);
+
+      //expect the start and end dates are stored in db are correct
+      expect(ooo[0].start).toEqual("2024-10-01T00:00:00.000Z");
+      expect(ooo[0].end).toEqual("2024-10-01T23:59:59.999Z");
+    });
+
+    it("should successfully create outofoffice with redirect for logged-in user.", async () => {
+      const teams = [
+        {
+          id: 1,
+          name: "Team 1",
```

Comment:
The test is more like a unit test than an integration test. We need an integration test for this type of complex handler change. If you really want this test, then move it to a different file location. Also, the test is too big. I suggested before that you only test the current changes you made because we already have end-to-end tests that cover the functionality so far. There's no point in testing similar functionality using two types of tests